source/2_platform/rex_windows/src/gui_application.cpp

```diff
@@ -126,14 +125,21 @@ namespace rex
 
       void subscribe_window_events()
       {
-        event_system::subscribe(event_system::EventType::WindowClose, [this](const event_system::Event& /*evt*/) { m_app_instance->quit(); });
+        event_system::subscribe(event_system::EventType::WindowClose,
+                                [this](const event_system::Event& /*evt*/)
+                                {
+                                  rex::event_system::Event ev {};
+                                  ev.type = rex::event_system::EventType::QuitApp;
+                                  rex::event_system::fire_event(ev);
+                                });
```

Comment:
Just as a reminder dunno if this is already in place but we should queue up events not call them directly this is dangerous

Reply:
do we have an issue for this?
the code in here shouldn't change imo, other than cleaning up how to fire an event cause it's 3 lines of code now while it should just be 1.
on backend site it should put this in a different queue than the one it's currently processing

Reply:
Is there an issue for the queue?

Reply:
Yes